.github/workflows/linux.yml

```diff
@@ -760,16 +760,114 @@ jobs:
           path: ${{ env.EXTENSION_BUILD_DIR }}/*.whl
           if-no-files-found: 'error'
 
-  GPU_Stub:
+  GPU:
+    name: GPU ${{ matrix.TEST_TYPE }} Tests
     needs: [ Build, Smart_CI ]
-    runs-on: ubuntu-latest
     if: fromJSON(needs.smart_ci.outputs.affected_components).GPU
+    timeout-minutes: 80
+    runs-on: [ self-hosted, gpu ]
+    strategy:
+      max-parallel: 2
+      fail-fast: false
+      matrix:
+        TEST_TYPE: ['unit', 'func']
+    container:
+      image: ubuntu:20.04
+      options: --device /dev/dri:/dev/dri --group-add 109 --group-add 44
+      volumes:
+        - /dev/dri:/dev/dri
+    defaults:
+      run:
+        shell: bash
+    env:
+      DEBIAN_FRONTEND: noninteractive # to prevent apt-get from waiting user input
+      INSTALL_DIR: ${{ github.workspace }}/install
+      INSTALL_TEST_DIR: ${{ github.workspace }}/install/tests
+      GTEST_PARALLEL_SCRIPT: ${{ github.workspace }}/gtest_parallel.py
     steps:
-      - name: GPU stub
+      - name: Download OpenVINO package
+        uses: actions/download-artifact@v4
+        with:
+          name: 'openvino_package'
+          path: ${{ env.INSTALL_DIR }}
+
+      - name: Download OpenVINO tests package
+        uses: actions/download-artifact@v4
+        with:
+          name: 'openvino_tests'
+          path: ${{ env.INSTALL_TEST_DIR }}
+
+      # Needed as ${{ github.workspace }} is not working correctly when using Docker
+      - name: Setup Variables
         run: |
-          echo "This is only a stub to collect statistics of GPU runs filtered by Smart CI.
-          It will help us to estimate hardware requirements"
-        shell: bash
+          echo "INSTALL_DIR=$GITHUB_WORKSPACE/install" >> "$GITHUB_ENV"
+          echo "INSTALL_TEST_DIR=$GITHUB_WORKSPACE/install/tests" >> "$GITHUB_ENV"
+          echo "GTEST_PARALLEL_SCRIPT=$GITHUB_WORKSPACE/gtest_parallel.py" >> "$GITHUB_ENV"
+
+      - name: Extract OpenVINO packages
+        run: |
+          pushd $INSTALL_DIR
+            tar -xzf openvino_package.tar.gz -C $INSTALL_DIR
+          popd
+          pushd $INSTALL_TEST_DIR
+            tar -xzf openvino_tests.tar.gz -C $INSTALL_DIR
+          popd
+
+      - name: Install dependencies (Linux)
+        run: |
+          $INSTALL_DIR/install_dependencies/install_openvino_dependencies.sh -c=core -c=dev -c=gpu -y
+
+          apt-get update && apt-get install -y wget software-properties-common ca-certificates gpg-agent tzdata
+        env:
+          DEBIAN_FRONTEND: noninteractive # to prevent apt-get from waiting user input
+          TZ: "Europe/London" # to prevent tzdata from waiting user input
+
+      - name: Setup Python ${{ env.PYTHON_VERSION }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+
+      - name: Get gtest-parallel script
+        run: wget https://raw.githubusercontent.com/google/gtest-parallel/master/gtest_parallel.py
```

Comment:
Why we don't use our script? it should work faster and exclude additional download dependencies

Reply:
I had trouble with running the GPU tests via `run_parallel.py`: https://github.com/openvinotoolkit/openvino/actions/runs/8572517506/job/23495935541, it seemed like it was running via single worker for unittests, though, it worked for the functional tests.

Additionally, it reported the `missing` tests, e.g.: https://github.com/openvinotoolkit/openvino/actions/runs/8572517506/job/23495935801

Also, for some reason, there were >300 failing functional tests when using the `run_parallel.py` script while no failing with `gtest_parallel.py`: https://github.com/openvinotoolkit/openvino/actions/runs/8572517506/job/23495935801.

It requires the installation of dependencies from `functional_test_utils/layer_tests_summary/requirements.txt`, the one from `gtest` uses only Python's standard library.

Reply:
I see, the root cause in filters, let's fix it separately. We would need to fix that, because it will significantly reduce execution time